DataFormats/Common/interface/MultiCollection.h

```diff
@@ -0,0 +1,63 @@
+// Author: Felice Pantaleo (CERN), 2025, felice.pantaleo@cern.ch
+#ifndef DataFormats_Common_MultiCollection_h
+#define DataFormats_Common_MultiCollection_h
+
+#include <ranges>
+#include <span>
+#include <utility>
+#include <vector>
+
+#include "DataFormats/Common/interface/RefProd.h"
+#include "MultiSpan.h"
+
+namespace edm {
+
+  /**
+ * @brief Lightweight persistent holder for several `edm::RefProd<Collection>`
+ *        objects.
+ *
+ * Only the vector of `RefProd`s is stored on disk.  No transient caches or
+ * synchronisation primitives live inside the class, so it is trivially
+ * movable.  Consumers obtain a flat view with `makeFlatView()`, which assembles
```

Comment:
I'd avoid the phrase "trivially movable", because while `std::vector` is efficiently movable, the move operations are not trivial (both `std::is_trivially_move_constructible()` and `std::is_trivially_move_assignable()` return `false` for `std::vector<T>`).